File: 560-subarray-sum-equals-k/subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        Map <Long,Integer> mp=new HashMap<>();
        mp.put(0L,1);
        int res=0;
        long ps=0;
        for (int n:nums){
            ps+=n;
            if (mp.containsKey(ps-k)) res+=mp.get(ps-k);
            mp.put(ps,mp.getOrDefault(ps,0)+1);
        }
        return res;
    }
}

File: 560-subarray-sum-equals-k/subarray-sum-equals-k.py
class Solution(object):
    def subarraySum(self, nums, k):
        res=0
        ps=0
        mp={0:1}

        for i in range(len(nums)):
            ps+=nums[i]
            if ps-k in mp:
                res+=mp.get(ps-k)
            mp[ps]=mp.get(ps,0)+1
        return res